scripts/player/player.gd

```diff
@@ -2,15 +2,31 @@ extends Entity
 
 class_name Player
 
+signal player_dead
```

Comment:
I think we should follow Godot naming pattern for signals and name it something like `on_player_death`.